corporate/lib/stripe.py

```diff
@@ -1170,6 +1189,50 @@ def set_required_plan_tier(self, required_plan_tier: int) -> str:
             plan_tier_name = CustomerPlan.name_from_tier(new_plan_tier)
         return f"Required plan tier for {self.billing_entity_display_name} set to {plan_tier_name}."
 
+    def create_fixed_price_plan(self, fixed_price: int) -> str:
+        customer = self.get_customer()
+        if customer is None:
+            customer = self.update_or_create_customer()
+
+        if customer.required_plan_tier is None:
+            raise SupportRequestError("Required plan tier should not be set to None")
+        required_plan_tier_name = CustomerPlan.name_from_tier(customer.required_plan_tier)
+
+        fixed_price_cents = fixed_price * 100
+        plan = get_current_plan_by_customer(customer)
+        if plan is not None:
+            fixed_price_plan = CustomerPlan.objects.create(
+                customer=customer,
+                automanage_licenses=True,
+                charge_automatically=plan.charge_automatically,
+                fixed_price=fixed_price_cents,
+                # 'billing_cycle_anchor' will be decided during /upgrade.
+                billing_cycle_anchor=timezone_now() - timedelta(weeks=52),
+                billing_schedule=CustomerPlan.BILLING_SCHEDULE_ANNUAL,
+                tier=customer.required_plan_tier,
+                status=CustomerPlan.NEVER_STARTED,
+            )
+            if self.check_plan_tier_is_billable(plan.tier):
+                assert plan.end_date is not None
+                fixed_price_plan.billing_cycle_anchor = plan.end_date
+                fixed_price_plan.save(update_fields=["billing_cycle_anchor"])
+                plan.status = CustomerPlan.SWITCH_PLAN_TIER_AT_PLAN_END
+                plan.save(update_fields=["status"])
+                return f"Fixed price {required_plan_tier_name} plan scheduled to start on {plan.end_date}."
+            else:
+                return f"Customer can now buy a fixed price {required_plan_tier_name} plan."
+        CustomerPlan.objects.create(
+            customer=customer,
+            automanage_licenses=True,
+            fixed_price=fixed_price_cents,
+            # 'billing_cycle_anchor' will be decided during /upgrade.
+            billing_cycle_anchor=timezone_now() - timedelta(weeks=52),
+            billing_schedule=CustomerPlan.BILLING_SCHEDULE_ANNUAL,
+            tier=customer.required_plan_tier,
+            status=CustomerPlan.NEVER_STARTED,
+        )
```

Comment:
This is almost the same as the value created above; I feel like it might be nicer to not repeat `CustomerPlan.objects.create(customer=customer,` -- I think there's just 1-2 parameters that would need an `if plan is None` conditional to set.

Reply:
Yeah, updated code to avoid repetition.